src/compas_eve/codecs.py

```diff
@@ -0,0 +1,102 @@
+"""
+********************************************************************************
+compas_eve.codecs
+********************************************************************************
+
+.. currentmodule:: compas_eve.codecs
+
+
+Classes
+=======
+
+.. autosummary::
+    :toctree: generated/
+    :nosignatures:
+
+    ProtobufMessageCodec
+
+"""
+
+from compas_eve.core import MessageCodec
+
+__all__ = ["ProtobufMessageCodec"]
+
+
+try:
+    import compas_pb
+
+    COMPAS_PB_AVAILABLE = True
+except ImportError:
+    COMPAS_PB_AVAILABLE = False
+
+
+class ProtobufMessageCodec(MessageCodec):
+    """Protocol Buffers codec for message serialization.
+
+    This codec uses the compas_pb package to encode and decode message data
+    using Protocol Buffers binary format. It can handle Message objects,
+    COMPAS Data objects, and regular dictionaries.
+
+    Note
+    ----
+    This codec requires the ``compas_pb`` package to be installed.
+    If ``compas_pb`` is not available, attempting to encode or decode
+    will raise an ImportError.
+    """
+
+    def __init__(self):
+        super(ProtobufMessageCodec, self).__init__()
+        if not COMPAS_PB_AVAILABLE:
+            raise ImportError(
+                "The ProtobufMessageCodec requires 'compas_pb' to be installed. "
+                "Please install it with: pip install compas_pb"
+            )
+
+    def encode(self, message):
+        """Encode a message to Protocol Buffers binary format.
+
+        Parameters
+        ----------
+        message : :class:`Message` or dict or object
+            Message to encode. Can be a Message instance, a dict, or
+            an object implementing the COMPAS data framework.
+
+        Returns
+        -------
+        bytes
+            Protocol Buffers binary representation of the message.
+        """
+        if not COMPAS_PB_AVAILABLE:
+            raise ImportError(
+                "The ProtobufMessageCodec requires 'compas_pb' to be installed. "
+                "Please install it with: pip install compas_pb"
+            )
+        # Extract data from the message
+        try:
+            data = message.data
+        except (KeyError, AttributeError):
+            try:
+                data = message.__data__
+            except (KeyError, AttributeError):
+                data = dict(message)
+        return compas_pb.encode(data)
```

Comment:
We should get `compas_pb` to byte-encode the whole message, so I guess something like:

```suggestion
        return compas_pb.pb_dump_bts(message)
```